modules/messages/src/lib.rs

```diff
@@ -191,6 +190,42 @@ pub mod pallet {
 		type OperatingModeStorage = PalletOperatingMode<T, I>;
 	}
 
+	#[pallet::hooks]
+	impl<T: Config<I>, I: 'static> Hooks<BlockNumberFor<T>> for Pallet<T, I>
+	where
+		u32: TryFrom<<T as frame_system::Config>::BlockNumber>,
+	{
+		fn on_idle(_block: T::BlockNumber, mut remaining_weight: Weight) -> Weight {
+			// we'll need at least to read outbound lane state, kill a message and update lane state
+			let db_weight = T::DbWeight::get();
+			if remaining_weight.all_lte(db_weight.reads_writes(1, 2)) {
+				return Weight::zero()
+			}
+
+			// messages from lane with index `i` in `ActiveOutboundLanes` are pruned when
+			// `System::block_number() % lanes.len() == i`. Otherwise we need to read lane states on
+			// every block, wasting the whole `remaining_weight` for nothing and causing starvation
+			// of the last lane pruning
+			let active_lanes = T::ActiveOutboundLanes::get();
+			let active_lanes_len = (active_lanes.len() as u32).into();
+			let active_lane_index = u32::unique_saturated_from(
+				frame_system::Pallet::<T>::block_number() % active_lanes_len,
+			);
+			let active_lane_id = active_lanes[active_lane_index as usize];
+
+			// first db read - outbound lane state
+			let mut active_lane = outbound_lane::<T, I>(active_lane_id);
+			let original_remaining_weight = remaining_weight;
+			remaining_weight -= db_weight.reads(1);
+			// and here we'll have writes. It is safe to do regular subtraction, since
+			// `prune_messages` respects the `remaining_weight`
+			remaining_weight -= active_lane.prune_messages(db_weight, remaining_weight);
+
+			// safe to use regular subtraction, since we respect the `remaining_weight`
+			original_remaining_weight - remaining_weight
```

Comment:
nit:
could simplify this to be smth like

```suggestion
			// first db read - outbound lane state
			let mut active_lane = outbound_lane::<T, I>(active_lane_id);
			let mut used_weight = db_weight.reads(1);
			// and here we'll have writes
			used_weight -= active_lane.prune_messages(db_weight, remaining_weight);

			// we already checked we have enough `remaining_weight` to cover this `used_weight`
			used_weight
```

Reply:
Thanks! I've also have been brave enough on Friday to leave this function untested && I admit I was wrong. So some of typos/issues are also fixed here. Additionally, I've moved lane check to the `send_message` function - now it check that the message is sent using one of lanes from `ActiveOutboundLanes` set.